src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php

```diff
@@ -277,6 +279,23 @@ public function commit(): bool
         return $ok;
     }
 
+    /**
+     * @throws BadMethodCallException When the item pool is not a NamespacedPoolInterface
+     */
+    public function withSubNamespace(string $namespace): static
+    {
+        if (!$this->pool instanceof NamespacedPoolInterface) {
+            throw new BadMethodCallException(\sprintf('Cannot call "%s::withSubNamespace()": this class doesn\'t implement "%s".', get_debug_type($this->pool), NamespacedPoolInterface::class));
+        }
+
+        $knownTagVersions = &$this->knownTagVersions; // ensures clones share the same array
```

Comment:
shouldn't the reference be assigned in the clone as well ? I don't see how taking a reference in a local variable will ensure sharing.

Reply:
that's the magic with reference: when cloning an object (or copying an array BTW), references remain references.
Because `$this->knownTagVersions` is a reference, `$clone->knownTagVersions` will also be one - to the same memory space.

Reply:
oh, I was not aware that cloning preserved references (I never investigated how references interact with cloning objects until now).

Reply:
@stof I think it's about the copy-on-write PHP mechanism, not about specifically about cloned objects

https://www.phpinternalsbook.com/php5/zvals/memory_management.html